src/notebooks/controllers/controllerDefaultService.ts

```diff
@@ -0,0 +1,125 @@
+import { inject, injectable } from 'inversify';
+import { NotebookDocument } from 'vscode';
+import { isPythonNotebook } from '../../kernels/helpers';
+import { ServerConnectionType } from '../../kernels/jupyter/launcher/serverConnectionType';
+import { IVSCodeNotebook } from '../../platform/common/application/types';
+import { InteractiveWindowView, JupyterNotebookView, PYTHON_LANGUAGE } from '../../platform/common/constants';
+import { IDisposableRegistry, IsWebExtension, Resource } from '../../platform/common/types';
+import { getNotebookMetadata } from '../../platform/common/utils';
+import { IInterpreterService } from '../../platform/interpreter/contracts';
+import { traceInfoIfCI, traceVerbose, traceDecoratorVerbose, traceError } from '../../platform/logging';
+import { createActiveInterpreterController } from './helpers';
+import {
+    IControllerDefaultService,
+    IControllerLoader,
+    IControllerRegistration,
+    IVSCodeNotebookController
+} from './types';
+
+@injectable()
+export class ControllerDefaultService implements IControllerDefaultService {
+    private get isLocalLaunch(): boolean {
+        return this.serverConnectionType.isLocalLaunch;
+    }
+    constructor(
+        @inject(IControllerRegistration) private readonly registration: IControllerRegistration,
+        @inject(IControllerLoader) private readonly loader: IControllerLoader,
+        @inject(IInterpreterService) private readonly interpreters: IInterpreterService,
+        @inject(IVSCodeNotebook) private readonly notebook: IVSCodeNotebook,
+        @inject(IDisposableRegistry) readonly disposables: IDisposableRegistry,
+        @inject(ServerConnectionType) private readonly serverConnectionType: ServerConnectionType,
+        @inject(IsWebExtension) private readonly isWeb: boolean
+    ) {}
+    public async computeDefaultController(
+        resource: Resource,
+        viewType: typeof JupyterNotebookView | typeof InteractiveWindowView
+    ): Promise<IVSCodeNotebookController | undefined> {
+        if (this.isLocalLaunch) {
+            traceInfoIfCI('CreateActiveInterpreterController');
+            return createActiveInterpreterController(viewType, resource, this.interpreters, this.registration);
+        } else {
+            traceInfoIfCI('CreateDefaultRemoteController');
+            const notebook =
+                viewType === JupyterNotebookView
+                    ? this.notebook.notebookDocuments.find((item) => item.notebookType === viewType)
```

Comment:
This would be wrong if we have multiple notebooks, we should look for the Uri as well.
I think passin g the notebook as a arg would fix this issue.

Reply:
This is the exact code as we already had. I just copied it.

Reply:
Might have to change that then, looks like a bug

Reply:
Yeah sure looks like it especially since it uses the notebook metadata. I'll add an issue.

Reply:
https://github.com/microsoft/vscode-jupyter/issues/10542